SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentReader.cpp

```diff
@@ -176,7 +176,27 @@ void ComponentReader::ApplySchemaObject(Schema_Object* ComponentObject, UObject&
 		// Scoped to exclude OnRep callbacks which are already tracked per OnRep function
 		SCOPE_CYCLE_COUNTER(STAT_ReaderApplyPropertyUpdates);
 
-		for (uint32 FieldId : UpdatedIds)
+		// If we are applying initial data, they must have come from a ComponentData (as it currently stands).
+		// ComponentData will be missing fields if they are completely empty (options, lists, and maps).
+		// However, we still want to apply this empty data, so we need to reconstruct the full
+		// list of field IDs for that component type (Data, OwnerOnly).
+		const TArray<Schema_FieldId>& IdsToIterate = bIsInitialData ?
+			ClassInfoManager->GetFieldIdsByComponentId(ComponentId, *Replicator->RepLayout.Get()) : UpdatedIds;
+#if DO_GUARD_SLOW
+		if (bIsInitialData)
+		{
+			auto CheckSubsetLambda = [](const TArray<Schema_FieldId>& Subset, const TArray<Schema_FieldId>& Superset) {
+				for (Schema_FieldId Field : Subset) {
+					if (!Superset.Contains(Field)) return false;
```

Comment:
I know, `{}`.
But it's so small! Doesn't algo or something have a subset check?

Reply:
Worth checing and if it's like the stl one it will need to be on sorted fields which I think this is.
Otherwise I'd still split the lines.

But my bigger question is why make this check.